dandi/tests/fixtures.py

```diff
@@ -296,75 +297,56 @@ def upload(self, paths=None, **kwargs) -> None:
+            # validate under dandischema.
             "contributor": [
                 {
                     "schemaKey": "Person",
                     "name": "Wodder, John",
                     "roleName": ["dcite:Author", "dcite:ContactPerson"],
                 }
             ],
-            "license": ["spdx:CC0-1.0"],
-            "manifestLocation": ["https://github.com/dandi/dandi-cli"],
         },
     )
-    dandiset_id = d.identifier
-    dspath = tmp_path_factory.mktemp("text_dandiset")
-    (dspath / dandiset_metadata_file).write_text(f"identifier: '{dandiset_id}'\n")
-    (dspath / "file.txt").write_text("This is test text.\n")
-    (dspath / "subdir1").mkdir()
-    (dspath / "subdir1" / "apple.txt").write_text("Apple\n")
-    (dspath / "subdir2").mkdir()
-    (dspath / "subdir2" / "banana.txt").write_text("Banana\n")
-    (dspath / "subdir2" / "coconut.txt").write_text("Coconut\n")
-    td = SampleDandiset(
+    dspath = tmp_path_factory.mktemp("dandiset")
+    (dspath / dandiset_metadata_file).write_text(f"identifier: '{d.identifier}'\n")
+    return SampleDandiset(
         api=local_dandi_api,
         dspath=dspath,
         dandiset=d,
-        dandiset_id=dandiset_id,
-        upload_kwargs={"allow_any_path": True},
+        dandiset_id=d.identifier,
     )
-    td.upload()
-    return td
 
 
 @pytest.fixture()
-def zarr_dandiset(local_dandi_api, monkeypatch, tmp_path_factory):
-    d = local_dandi_api.client.create_dandiset(
-        "Zarr Dandiset",
-        {
-            "schemaKey": "Dandiset",
-            "name": "Zarr Dandiset",
-            "description": "A test Zarr Dandiset",
-            "contributor": [
-                {
-                    "schemaKey": "Person",
-                    "name": "Wodder, John",
-                    "roleName": ["dcite:Author", "dcite:ContactPerson"],
-                }
-            ],
-            "license": ["spdx:CC0-1.0"],
-            "manifestLocation": ["https://github.com/dandi/dandi-cli"],
-        },
-    )
-    dandiset_id = d.identifier
-    dspath = tmp_path_factory.mktemp("zarr_dandiset")
-    (dspath / dandiset_metadata_file).write_text(f"identifier: '{dandiset_id}'\n")
-    zarr.save(dspath / "sample.zarr", np.arange(1000), np.arange(1000, 0, -1))
-    td = SampleDandiset(
-        api=local_dandi_api,
-        dspath=dspath,
-        dandiset=d,
-        dandiset_id=dandiset_id,
+def text_dandiset(new_dandiset):
+    (new_dandiset.dspath / "file.txt").write_text("This is test text.\n")
+    (new_dandiset.dspath / "subdir1").mkdir()
+    (new_dandiset.dspath / "subdir1" / "apple.txt").write_text("Apple\n")
+    (new_dandiset.dspath / "subdir2").mkdir()
+    (new_dandiset.dspath / "subdir2" / "banana.txt").write_text("Banana\n")
+    (new_dandiset.dspath / "subdir2" / "coconut.txt").write_text("Coconut\n")
+    new_dandiset.upload_kwargs["allow_any_path"] = True
+    new_dandiset.upload()
+    return new_dandiset
+
+
+@pytest.fixture()
```

Comment:
I wonder if both `text_` and `zarr_` dandisets should become `(scope="session")` since they look like the ones to be reused for read-only purposes. And have some dedicated `new_zarr_dandiset` (and similar `new_text_dandiset`) with scope to the test for the use cases where needed to mutate?

Reply:
I think that would only be a good idea if we could somehow prevent the read-only Dandiset fixtures from being modified (both on disk and on the server); otherwise, it'd be too easy to accidentally use the wrong fixture and cause a test failure that ends up being hard to debug.  I don't think the repeated recreating of Dandisets slows the tests down all that much.

Reply:
yeah, inability to guarantee immutability also worried me.  I even was reminded today about https://github.com/pytest-dev/pytest/issues/5044 after I thought that such "presumably a read only use of dandist" could be tested in fixture "turndown" code -- then we could catch cases where a test unexpectedly modifies a dandiset.  I was thinking that there could be 
- code which creates a new dandiset
- test scoped fixture which creates new dandiset only if doesn't exist, yields to the test, verifies that test did not change it.  If it did -- destroys pre-created dandiset (so it would be created a new on next run) and `.fail`s the test with a message that immutability assumption was modified.

But I guess it might be all too much for this PR, so I am ok to proceed with this PR as is unless you see interest in above.

Reply:
Let's stick with the PR as it is now.